ldap/src/main/java/org/springframework/security/ldap/DefaultSpringSecurityContextSource.java

```diff
@@ -74,7 +78,12 @@ public DefaultSpringSecurityContextSource(String providerUrl) {
 			rootDn = (rootDn != null) ? rootDn : urlRootDn;
 		}
 		setUrls(urls.toArray(new String[0]));
-		setBase(rootDn);
+		try {
+			setBase((rootDn != null) ? URLDecoder.decode(rootDn, StandardCharsets.UTF_8.toString()) : null);
+		}
+		catch (UnsupportedEncodingException ex) {
+			throw new IllegalArgumentException("Unable to URL decode base dn: " + rootDn);
```

Comment:
I think `IllegalArgumentException` fits better when the user's input is the problem. However, `UnsupportedEncodingException` appears to be a problem with the application itself. Maybe `IllegalStateException` would be better.